api-lib/types.d.ts

```diff
@@ -0,0 +1,75 @@
+import { GraphQLTypes } from '../src/lib/gql/zeusHasuraAdmin';
+
+type OperationTypes = 'INSERT' | 'UPDATE' | 'DELETE' | 'MANUAL';
+
+// This is not an exhaustive list of all object and array relation types,
+// and should be added to
+// as more relationships are encountered in the GQL api.
+// Only scalar types are returned by event triggers, so this is needed
+// to strip out faulty types from the autogenerated zeus types
+//
+// Maybe one day typescript will allow us to pick based on the type of
```

Comment:
This seems like a pain! trying to understand it. This led me to some docs that helped me understand TS a little bit better tho. I think Partial can help me with some of my other struggles.